tests/Components/PluginConfigTest.php

```diff
@@ -80,40 +80,66 @@ public function testPushIsForwardedToConfigRepository()
         $this->getPluginConfig()->push('testKey', 'testValue');
     }
 
-    public function testGetShopKeyReturnsCorrectKeyBasedOnCurrentLanguage()
+    public function testGetShopKeyReturnsNullIfShopKeyIsEmpty()
     {
         $this->configRepository->expects($this->once())
             ->method('get')
             ->with(Plugin::CONFIG_SHOPKEY, '')
-            ->willReturn("de:ABC123\nen:DEF456");
+            ->willReturn('');
 
-        $this->sessionStorageService->expects($this->once())->method('getLang')->willReturn('de');
+        $this->sessionStorageService->expects($this->never())->method('getLang');
 
-        $this->assertSame($this->getPluginConfig()->getShopKey(), 'ABC123');
+        $this->assertSame($this->getPluginConfig()->getShopKey(), null);
     }
 
-    public function testGetShopKeyReturnsNullWhenLanguageIsNotConfiguredForShop()
+    public function getShopKeyProvider()
     {
-        $this->configRepository->expects($this->once())
-            ->method('get')
-            ->with(Plugin::CONFIG_SHOPKEY, '')
-            ->willReturn("de:ABC123\nen:DEF456");
-
-        $this->sessionStorageService->expects($this->once())->method('getLang')->willReturn('jp');
-
-        $this->assertNull($this->getPluginConfig()->getShopKey());
+        return [
+            'It returns correct key based on current language' => [
+                "de:ABC123\nen:DEF456",
+                'de',
+                'ABC123'
+            ],
+            'It returns null when no generic shopkey and no shopkey for current language are configured 1' => [
+                "de:ABC123\njp:DEF456",
+                'en',
+                null
+            ],
+            'It returns null when no generic shopkey and no shopkey for current language are configured 2' => [
+                "en:ABC123",
+                'de',
+                null
+            ],
+            'It returns generic shopkey if no shopkey for current language exists 1' => [
+                "ABC123\nde:DEF456",
+                'en',
+                'ABC123'
+            ],
+            'It returns generic shopkey if no shopkey for current language exists 2' => [
+                "en:ABC123\nDEF456",
+                'de',
+                'DEF456'
+            ]
+        ];
     }
 
-    public function testGetShopKeyReturnsNullIfShopKeyIsEmpty()
+    /**
+     * @dataProvider getShopKeyProvider
+     *
+     * @param string $configRepositoryGetReturnValue
+     * @param string $currentLang
+     * @param string|null $returnValue
+     */
+    public function testGetShopKey($configRepositoryGetReturnValue, $currentLang, $returnValue)
```

Comment:
`$returnValue` and `$configRepositoryGetReturnValue` are odd names, as a "return value" could be literally anything.

* `$configRepositoryGetReturnValue` => `$configuredShopkeys` (maybe even `$raw...`, to make it clear this value has not been touched yet)
* `$returnValue` => `$expectedShopkey`

Naming the variables what they contain, instead of where they will be used, hides implementation specific details, which makes this just way easier to read and to understand.